### PR TITLE
Fix syntax warning over comparison of literals using is.

### LINKED ISSUE #3066: fix SyntaxWarnings for 'is' comparison in test_util.py
## Expected behavior ##

Tests under Python 3.8 should not emit SyntaxWarnings.
```
SyntaxWarning: "is" with a literal. Did you mean "=="?
```

## Actual behavior ##

Many tests in test_util.py (and possibly elsewhere) use constructs like
```python
assert x is 1
```
As explained in https://adamj.eu/tech/2020/01/21/why-does-python-3-8-syntaxwarning-for-is-literal/, that might work by accident (i.e., in CPython) but is not correct. Instead it should be
```python
assert x == 1
```

## Code to reproduce the behavior ##

See *install_mda* in https://github.com/MDAnalysis/mdanalysis/pull/3061/checks?check_run_id=1520450635#step:9:2070
```
 build/bdist.linux-x86_64/egg/MDAnalysisTests/lib/test_util.py:1429: SyntaxWarning: "is" with a literal. Did you mean "=="?
  assert myfunc.foo is 0
build/bdist.linux-x86_64/egg/MDAnalysisTests/lib/test_util.py:1443: SyntaxWarning: "is" with a literal. Did you mean "=="?
  assert x[0] is 1
build/bdist.linux-x86_64/egg/MDAnalysisTests/lib/test_util.py:1444: SyntaxWarning: "is" with a literal. Did you mean "=="?
  assert myfunc.bar['test'][0] is 1
build/bdist.linux-x86_64/egg/MDAnalysisTests/lib/test_util.py:1450: SyntaxWarning: "is" with a literal. Did you mean "=="?
  assert myfunc.bar['test'][0] is 2
build/bdist.linux-x86_64/egg/MDAnalysisTests/lib/test_util.py:1490: SyntaxWarning: "is" with a literal. Did you mean "=="?
  assert x is 0
build/bdist.linux-x86_64/egg/MDAnalysisTests/lib/test_util.py:1502: SyntaxWarning: "is" with a literal. Did you mean "=="?
  assert x is 0
build/bdist.linux-x86_64/egg/MDAnalysisTests/lib/test_util.py:1556: SyntaxWarning: "is" with a literal. Did you mean "=="?
  assert x is 0
```

## Current version of MDAnalysis ##

- Which version are you using? (run `python -c "import MDAnalysis as mda; print(mda.__version__)"`) current
- Which version of Python (`python -V`)?
- Which operating system?

### PR DESCRIPTION
**Pull request by tirkarthi**
Fixes #3066 

Changes made in this Pull Request:

Fix syntax warning over using is.

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

File: package/MDAnalysis/analysis/legacy/x3dna.py
Comment (IAlibay) [inline package/MDAnalysis/analysis/legacy/x3dna.py:426]: Why is this being changed?

**Review by IAlibay — COMMENTED** [review]
Thanks for starting this @tirkarthi 

I'm going to leave the review for when CI returns so we can check this dealt with the warnings. Just dropping a single comment/question for now.

Also; please do remember to add yourself to AUTHORS and the author list in the CHANGELOG (I don't think this warrants a changelog entry though).

File: package/MDAnalysis/analysis/legacy/x3dna.py
Comment (tirkarthi) [reply package/MDAnalysis/analysis/legacy/x3dna.py:426]: This line generates a warning about invalid escape sequences. I have used pyupgrade to fix these warnings : https://github.com/asottile/pyupgrade#invalid-escape-sequences

```
find . -iname '*.py' | xargs -P4 -I{} python3.8 -Wall -m py_compile {}   
./testsuite/MDAnalysisTests/lib/test_util.py:1477: SyntaxWarning: "is" with a literal. Did you mean "=="?
  assert myfunc.foo is 0
./testsuite/MDAnalysisTests/lib/test_util.py:1491: SyntaxWarning: "is" with a literal. Did you mean "=="?
  assert x[0] is 1
./testsuite/MDAnalysisTests/lib/test_util.py:1492: SyntaxWarning: "is" with a literal. Did you mean "=="?
  assert myfunc.bar['test'][0] is 1
./testsuite/MDAnalysisTests/lib/test_util.py:1498: SyntaxWarning: "is" with a literal. Did you mean "=="?
  assert myfunc.bar['test'][0] is 2
./testsuite/MDAnalysisTests/lib/test_util.py:1539: SyntaxWarning: "is" with a literal. Did you mean "=="?
  assert x is 0
./testsuite/MDAnalysisTests/lib/test_util.py:1551: SyntaxWarning: "is" with a literal. Did you mean "=="?
  assert x is 0
./testsuite/MDAnalysisTests/lib/test_util.py:1605: SyntaxWarning: "is" with a literal. Did you mean "=="?
  assert x is 0
./package/MDAnalysis/analysis/legacy/x3dna.py:426: DeprecationWarning: invalid escape sequence \A
  ax.set_ylabel("{0!s} ($\AA$)".format((param)))
```

**Comment by lilyminium** [comment]
Thank you @tirkarthi, this is very welcome! Could you please add yourself to the AUTHORS file and update the CHANGELOG?

**Review by orbeckst — CHANGES_REQUESTED** [review]
@tirkarthi thank you for the fixes.

Please add yourself to AUTHORS and update the CHANGELOG under _Fixes_ (because you closed an issue).

There's an AppVeyor test that's failing. We retired AppVeyor in with #3065 (PR #3073) so could you please merge the latest develop branch into your branch and push? This should get rid of AppVeyor tests.

**Comment by tirkarthi** [comment]
Thanks @orbeckst , I just updated the files and rebased my branch with latest develop branch

**Review by lilyminium — APPROVED** [review]

**Comment by lilyminium** [comment]
Thanks @tirkarthi!

File: package/CHANGELOG
Comment (lilyminium) [inline package/CHANGELOG:78]: Could you please just add your Github username to the author list above @tirkarthi?

Comment (tirkarthi) [reply package/CHANGELOG:78]: Thanks, added.

**Review by orbeckst — APPROVED** [review]

**Comment by orbeckst** [comment]
Thanks @tirkarthi – I'll merge once the CI comes back green from the merge conflicts that I resolved.

**Comment by tirkarthi** [comment]
Thank you all for the review and merge.

**Pull request merged**
